sender/batch.go

```diff
@@ -60,19 +60,29 @@ This means that:
 3. Send() will only block if two channels are full.
 */
 type Batch struct {
-	Next      skogul.SenderRef `doc:"Sender that will receive batched metrics"`
-	Interval  skogul.Duration  `doc:"Flush the bucket after this duration regardless of how full it is"`
-	Threshold int              `doc:"Flush the bucket after reaching this amount of metrics"`
-	allocSize int
-	ch        chan *skogul.Container
-	once      sync.Once
-	metrics   int
-	timer     *time.Timer
-	cont      *skogul.Container
-	out       chan *skogul.Container
+	Next       skogul.SenderRef `doc:"Sender that will receive batched metrics"`
+	Interval   skogul.Duration  `doc:"Flush the bucket after this duration regardless of how full it is"`
+	Threshold  int              `doc:"Flush the bucket after reaching this amount of metrics"`
+	allocSize  int
+	ch         chan *skogul.Container
+	once       sync.Once
+	metrics    int
+	timer      *time.Timer
+	cont       *skogul.Container
+	out        chan *skogul.Container
+	numThreads int
+}
+
+func (bat *Batch) fillUninitialized() {
+	if bat.numThreads == 0 {
+		bat.numThreads = runtime.NumCPU()
+	}
 }
 
 func (bat *Batch) setup() {
+
+	bat.fillUninitialized()
+
```

Comment:
I think this should be done inline in the setup func instead of splitting it out into its own function.

Reply:
Sounds fair, I would have incorporated this earlier, but I wasn't sure about your preferences. I'll fix this :)